sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/AutocompleteSyncTests.java

```diff
@@ -23,20 +41,59 @@
 
 public class AutocompleteSyncTests extends SearchTestBase {
     private static final String HOTELS_DATA_JSON = "HotelsDataArray.json";
+    private static final String INDEX_NAME = "azsearch-autocomplete-shared-instance";
 
+    private static SearchIndexClient searchIndexClient;
     private SearchClient client;
 
-    @Override
-    protected void beforeTest() {
-        super.beforeTest();
 
-        client = getSearchClientBuilder(createHotelIndex()).buildClient();
-        uploadDocumentsJson(client, HOTELS_DATA_JSON);
+
+    @BeforeAll
+    public static void setupClass() {
+        TestBase.setupClass();
+
+        if (TEST_MODE == TestMode.PLAYBACK) {
+            return;
+        }
+
+        Reader indexData = new InputStreamReader(Objects.requireNonNull(AutocompleteSyncTests.class
+            .getClassLoader()
+            .getResourceAsStream(HOTELS_TESTS_INDEX_DATA_JSON)));
+
+        try {
+            SearchIndex index = new ObjectMapper().readValue(indexData, SearchIndex.class);
+
+            Field searchIndexName = index.getClass().getDeclaredField("name");
+            AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+                searchIndexName.setAccessible(true);
+                return null;
+            });
+
+            searchIndexName.set(index, INDEX_NAME);
```

Comment:
Why can't we change the name directly in hotel json file but to modify here after serialization?
There are two other test classes which have the similar structure, SuggestSyncTests and SearchSyncTest which can do the similar stuff.

Reply:
Updated Search and Suggest test which are able to share an instance. The name is changed using reflection as many tests use this file so it won't be safe to make it a global name.